### PR TITLE
NGFC Flowsheet Improvements

### PR DESCRIPTION
**Pull request by AlexNoring**
# NGFC Flowsheet Improvements

Updates to the NGFC flowsheet to fix initialization failure
- Appropriate solver options are now being used for flowsheet initialization
- Improved model scaling
- Adjusted the initialization/solve sequence for the main flowsheet and the solver options for the final solve
- Reformer and power island sections can be now be build separately for debugging

----
Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

    I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
    I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

----
📚 Documentation preview 📚: https://idaes-examples--105.org.readthedocs.build/en/105/

### CODE REVIEW

File: idaes_examples/mod/power_gen/NGFC_flowsheet.py [hunk 1]
```diff
@@ -767,219 +768,180 @@ def scale_flowsheet(m):
...
-        if hasattr(unit, "isentropic_pressure"):
-            for t, c in unit.isentropic_pressure.items():
-                iscale.constraint_scaling_transform(c, 1e-5, overwrite=False)
-
-        if hasattr(unit, "isentropic"):
-            for t, c in unit.isentropic.items():
-                iscale.constraint_scaling_transform(c, 1e-1, overwrite=False)
-
+        # pressure changer
         if hasattr(unit, "isentropic_energy_balance"):
```
Comment (andrewlee94) [inline idaes_examples/mod/power_gen/NGFC_flowsheet.py:888]: Probably outside the scope of this PR, but are all these 'if' checks required? They make sense for a generic unit model where users have options over what terms might be included, but for a flowsheet example I would have thought most of these would be concretely defined (i.e. you know if it is here or not).

**Review by andrewlee94 — APPROVED** [review]
I don't see any major issues, however we do need to resolve the testing issue due to `PySimpleGUI`.

**Review by JavalVyas2000 — APPROVED** [review]

**Comment by JavalVyas2000** [comment]
@andrewlee94 should we move this example from archive to docs?

File: idaes_examples/archive/power_gen/ngfc/ngfc_flowsheet_src.ipynb [hunk 2]
```diff
@@ -167,29 +155,59 @@
...
+    "solver_ma57.options = {\n",
+    "    \"max_iter\": 200,\n",
+    "    \"tol\": 1e-7,\n",
+    "    \"bound_push\": 1e-5,\n",
+    "    \"linear_solver\": \"ma57\",\n",
+    "    \"OF_ma57_automatic_scaling\": \"yes\",\n",
+    "    \"nlp_scaling_method\": \"user-scaling\"\n",
+    "}\n",
+    "\n",
+    "# suppress warnings about missing scaling factors\n",
```
[the remark this answers is not in the document]
Comment (andrewlee94) [reply idaes_examples/archive/power_gen/ngfc/ngfc_flowsheet_src.ipynb:199]: Yes, we should suppress all these warnings in examples.

**Comment by andrewlee94** [comment]
@JavalVyas2000 I believe it was on the list of examples Miguel suggested we make public so the answer is yes (assuming we can get it working reliably). I'll leave it up to you and @AlexNoring as to whether that should be done in this PR or not - note that if we do that we should add more diagnostics checks as well (`assert_no_strucutral_issues` and `assert_no_numerical_issues`) to make sure the flowsheet is well posed and robust.

**Comment by AlexNoring** [comment]
Let's leave moving this example to docs for a separate PR.

**Pull request merged**